### PR TITLE
Issue 17941: Add oc new-build --push-secret option

### PR DESCRIPTION
**Pull request by damemi**
Adds a `--push-secret` option to `oc new-build` as requested in https://github.com/openshift/origin/issues/17941

fixes https://github.com/openshift/origin/issues/17941

### CODE REVIEW

File: pkg/oc/cli/cmd/newbuild.go [hunk 1]
```diff
@@ -114,6 +114,7 @@ func NewCmdNewBuild(name, baseName string, f *clientcmd.Factory, in io.Reader, o
 	cmd.Flags().StringSliceVar(&config.DockerImages, "docker-image", config.DockerImages, "Name of a Docker image to use as a builder.")
 	cmd.Flags().StringSliceVar(&config.Secrets, "build-secret", config.Secrets, "Secret and destination to use as an input for the build.")
 	cmd.Flags().StringVar(&config.SourceSecret, "source-secret", "", "The name of an existing secret that should be used for cloning a private git repository.")
+	cmd.Flags().StringVar(&config.PushSecret, "push-secret", "", "The name of an existing secret that should be used for pushing to a Docker repository if --to-docker and --to are set.")
```
Comment (bparees) [inline pkg/oc/cli/cmd/newbuild.go]: >  if --to-docker and --to are set

the way you've currently implemented this, the secret will be used regardless of whether --to/--to-docker are set.  I'd just write the help as "The name of an existing secret that should be used for pushing the output image"

Comment (damemi) [reply pkg/oc/cli/cmd/newbuild.go]: thanks, updated and added tests

**Comment by bparees** [comment]
@damemi can you add secret name validation for the new argument (and a test) like is being done here:
https://github.com/openshift/origin/pull/18429
?

(sorry, just figured we might as well be consistent w/ the other change that's going in there)

**Comment by damemi** [comment]
@bparees no problem, that makes sense. updated

**Comment by bparees** [comment]
thanks @damemi 
/lgtm

**Pull request merged**